main.tf

```
@@ -71,6 +71,84 @@ resource "aws_iam_user_policy" "quortex_bucket_rw" {
 EOF
 }
 
+# --- Roles and policies associated to irsa ---
+data "aws_caller_identity" "current" {
+  count = var.enable_irsa ? 1 : 0
+}
+
+resource "aws_iam_role" "aws_eks_irsa" {
+  for_each           = var.enable_irsa ? local.buckets : {}
```

This could be filtered depending on wether roles are enabled : 
```suggestion
  for_each           = var.enable_irsa ? {for b in var.buckets : b["name"] => b if b.role != null} : {}
```